l20n/migrate/test_context_real_examples.py

```diff
@@ -0,0 +1,415 @@
+        # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+        title  = Pobrane pliki
+        header = Twoje pobrane pliki
+        empty  = Brak pobranych plików
+
+        open-menuitem =
+            [html/label] Otwórz
+        retry-menuitem =
+            [html/label] Spróbuj ponownie
+        remove-menuitem =
+            [html/label] Usuń
+        pause-menuitem =
+            [html/label] Wstrzymaj
+        resume-menuitem =
+            [html/label] Wznów
+        cancel-menuitem =
+            [html/label] Anuluj
+        remove-all-menuitem =
+            [html/label] Usuń wszystko
+
+        delete-all-title   = Usuń wszystko
+        delete-all-message = { $num ->
+            [one] Usunąć pobrany plik?
+            [few] Usunąć { $num } pobrane pliki?
+           *[many] Usunąć { $num } pobranych plików?
+        }
+
+        download-state-downloading = Pobieranie…
+        download-state-canceled    = Anulowane
+        download-state-failed      = Nieudane
+        download-state-paused      = Wstrzymane
+        download-state-starting    = Rozpoczynanie…
+        download-size-unknown      = Nieznany rozmiar
+            ''')
+        }
+
+        self.assertDictEqual(
+            to_json(self.ctx.merge_changeset()),
+            expected
+        )
+
+    def test_merge_context_some_messages(self):
+        changeset = {
+            ('aboutDownloads.dtd', 'aboutDownloads.title'),
+            ('aboutDownloads.dtd', 'aboutDownloads.header'),
+            ('aboutDownloads.properties', 'downloadState.downloading'),
+            ('aboutDownloads.properties', 'downloadState.canceled'),
+        }
+
+        expected = {
+            'aboutDownloads.ftl': ftl_resource_to_json('''
+        # This Source Code Form is subject to the terms of the Mozilla Public
+        # License, v. 2.0. If a copy of the MPL was not distributed with this
+        # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+        title                      = Pobrane pliki
+        header                     = Twoje pobrane pliki
+        download-state-downloading = Pobieranie…
+        download-state-canceled    = Anulowane
+            ''')
+        }
+
+        self.assertDictEqual(
+            to_json(self.ctx.merge_changeset(changeset)),
+            expected
+        )
+
+
+class TestMergeAboutDialog(unittest.TestCase):
+    def setUp(self):
+        self.ctx = MergeContext(
+            lang='pl',
+            reference_dir='fixtures/en-US',
+            localization_dir='fixtures/pl'
+        )
+
+        self.ctx.add_reference('aboutDialog.ftl')
+        self.ctx.add_localization('aboutDialog.dtd')
+
+        self.ctx.add_transforms('aboutDialog.ftl', [
+            FTL.Entity(
+                id=FTL.Identifier('update-failed'),
+                value=CONCAT(
+                    COPY(
+                        SOURCE('aboutDialog.dtd', 'update.failed.start'),
+                    ),
+                    COPY('<a>'),
```

Comment:
I think I'd like to change the AST a bit and make `COPY` a subclass of `SOURCE`. The above would then look like this:

```python
value=CONCAT(
    COPY('aboutDialog.dtd', 'update.failed.start'),
    LITERAL('<a>'),
    ...
)
```